src/frontend/src/pages/index.js

```diff
@@ -90,17 +91,20 @@ export default function IndexPage() {
   }
 
   const getPostsCount = useCallback(async () => {
-    try {
-      const res = await fetch(`${telescopeUrl}/posts`, { method: 'HEAD' });
-      if (!res.ok) {
-        throw new Error(res.statusText);
+    if (update) {
```

Comment:
```js
if(!update) {
  return null;
}
...continue without indent
```

Reply:
Doesn't this way make it kind of messy with two return null statements?

```
const getPostsCount = useCallback(async () => {
    if (!shouldCheckForNewPosts) {
      return null;
    }
    try {
      const res = await fetch(`${telescopeUrl}/posts`, { method: 'HEAD' });
      if (!res.ok) {
        throw new Error(res.statusText);
      }
      setShouldCheckForNewPosts(false);
      return res.headers.get('x-total-count');
    } catch (error) {
      console.log(error);
    }
    return null;
  }, [telescopeUrl, shouldCheckForNewPosts]);
```

Reply:
Don't believe what your other profs told you, [Return Early, Return Often](https://blog.humphd.org/vocamus-1421/).